Add support for properties in COM source generator (#128869)
dotnet/runtime: docs/design/libraries/ComInterfaceGenerator/DefaultImplementedMembers.md

    // Zero vtable slots — managed sugar wrapping the two ABI methods above.
    double Value
    {
        get => ReadValue();
        set => WriteValue(value);
    }

    // Also zero vtable slots — a managed-only helper.
    double DoubleIt() => ReadValue() * 2;
}
```

DIM is the user-facing escape hatch for scenarios the canonical "method &rarr; one vtable slot" / "property &rarr; one or two adjacent vtable slots" rules cannot express, including:

* Wrapping ABI methods whose names do not match the canonical `get_X` / `set_X` accessor naming.
* Wrapping ABI methods that live on different vtable slots than two adjacent slots would imply.
* Adding helper methods that the wire ABI does not need to know about.

## Rules and diagnostics

* **All accessors must agree.** A property must have either all-abstract accessors (`int X { get; set; }`) or all-bodied accessors (`int X { get => …; set => …; }`). Mixing the two emits the error `SYSLIB1091` `PropertyAccessorsMustBeAllOrNothing`. (The C# compiler also rejects bare `get;` paired with a bodied `set { … }` with `CS0525`, because it interprets `get;` as an auto-property accessor and disallows auto-properties on interfaces.)

> * **`[MarshalUsing]` and `[MarshalAs]` on a DIM are a warning.** A DIM never participates in marshalling, so any marshal attribute on the DIM (the property itself, an accessor return, a method parameter, etc.) emits a `SYSLIB1091` warning `MarshalAttributeOnDefaultImplementedComInterfaceMember`. The DIM is still generated; the marshal attribute is silently ignored at the ABI layer.

Reviewer: This line didn't match my understanding, would this be more accurate or am I missing something?

```suggestion
* **`[MarshalUsing]` and `[MarshalAs]` on a DIM are a warning.** A DIM never participates in marshalling, so any marshal attribute on the DIM (the property itself, an accessor return, a method parameter, etc.) emits a `SYSLIB1091` warning `MarshalAttributeOnDefaultImplementedComInterfaceMember`. The code generator does not generate code for the DIM, so the presence of a marshal attribute is misleading.
```
